src/AzerothWarsCSharp.Source/Researches/Lordaeron/VeteranFootmen.cs

```diff
@@ -1,22 +1,24 @@
-// using AzerothWarsCSharp.MacroTools.FactionSystem;
-// using AzerothWarsCSharp.Source.Setup.FactionSetup;
-//
-// using static War3Api.Common;  using static AzerothWarsCSharp.MacroTools.GeneralHelpers;
-// {
-//   public class VeteranFootmen{
-//
-//   
-//     private const int RESEARCH_ID = FourCC("R00B");
-//   
-//
-//     private static void Research( ){
-//       LordaeronSetup.FactionLordaeron.ModObjectLimit(FourCC("hfoo"), -Faction.UNLIMITED)  ;//Footman
-//       LordaeronSetup.FactionLordaeron.ModObjectLimit(FourCC("h029"), Faction.UNLIMITED)   ;//Veteran Footman
-//     }
-//
-//     public static void Setup( ){
-//       RegisterResearchFinishedAction(RESEARCH_ID,  Research);
-//     }
-//
-//   }
-// }
+using AzerothWarsCSharp.MacroTools.FactionSystem;
+using AzerothWarsCSharp.Source.Setup.FactionSetup;
+using WCSharp.Events;
+using static War3Api.Common;
+
+namespace AzerothWarsCSharp.Source.Researches.Lordaeron
+{
+  /// <summary>
+  /// When Veteran Footman is researched, the researching player loses the ability to train Footmen, and gains the ability to train Veteran Footmen.
+  /// </summary>
+  public class VeteranFootmen
```

Comment:
Make this class static since it doesn't have any instance methods. 

```suggestion
  public static class VeteranFootmen
```